test/jdk/java/lang/ProcessBuilder/ArgCheck.java

```diff
@@ -0,0 +1,314 @@
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8282008
+ * @requires (os.family == "windows")
+ * @run main/othervm ArgCheck
+ * @summary Check invocation of exe and non-exe programs using ProcessBuilder
+ *      and arguments with spaces, backslashes, and simple quoting.
+ */
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Class to check invocation of java, .cmd, and vbs scripts with arguments and various quote cases.
+ * Can be run standalone to compare results with other Java versions.
+ */
+public class ArgCheck {
+
+    private static final Path SRC_DIR = Paths.get(System.getProperty("test.src", "."));
+    private static final Path WORK_DIR = Paths.get(System.getProperty("user.dir", "."));
+    private static final Path TEST_CLASSES = Paths.get(System.getProperty("test.classes", "."));
+
+    private static final String ECHO_CMD_PATH = WORK_DIR.resolve("EchoArguments.cmd").toString();
+    private static final String ECHO_VBS_PATH = WORK_DIR.resolve("EchoArguments.vbs").toString();
+
+    // Test argument containing both a space and a trailing backslash
+    // Depending on the mode the final backslash may act as an escape that may turn an added quote to a literal quote
+    private static final String SPACE_AND_BACKSLASH = "SPACE AND BACKSLASH\\";
+    private static final char DOUBLE_QUOTE = '"';
+    private static final char NEWLINE = '\n';
+    private static final char BACKSLASH = '\\';
+
+    private static final String AMBIGUOUS_PROP_NAME = "jdk.lang.Process.allowAmbiguousCommands";
+    private static final String AMBIGUOUS_PROP_VALUE = System.getProperty(AMBIGUOUS_PROP_NAME);
+    private static final Boolean AMBIGUOUS_PROP_BOOLEAN = AMBIGUOUS_PROP_VALUE == null ? null :
+                                                          Boolean.valueOf(!"false".equals(AMBIGUOUS_PROP_VALUE));
```

Comment:
Hello Roger,

Could these 2 lines be replaced by:

```
AMBIGUOUS_PROP_BOOLEAN = Boolean.getBoolean(AMBIGUOUS_PROP_NAME);
```

Reply:
Actually, never mind, I think I see why you did it this way. It looks like the test also wants to test wants to distinguish the case where that property may not have been set at all and the case where it might have been set to `false`. Using `Boolean.getProperty` wouldn't provide that distinction.